tools/clang/tools/dxclib/dxc.cpp

```diff
@@ -264,6 +264,14 @@ static void WriteDxcExtraOuputs(IDxcResult *pResult) {
   }
 }
 
+std::string getDependencyOutputFileName(llvm::StringRef inputFileName) {
+  size_t extensionOffset = inputFileName.rfind('.');
+  if (extensionOffset == llvm::StringRef::npos)
+    return inputFileName;
```

Comment:
So this will cause the dependency dump to be written back to the input file if it didn't have an extension?  Is that really what we intend?  It seems like we would instead want to set `extensionOffset` to `inputFileName.size()` so `.d` would be appended without removing any extension.

I think the desired behavior can be within in one expression:
```
return inputFileName.substr(0, inputFileName.rfind('.')).str() + ".d";
```

Reply:
Ah .. this was a mistake. Thank you for letting me know!